app/models/manageiq/providers/openstack/cloud_manager/auth_key_pair.rb

```diff
@@ -1,2 +1,33 @@
 class ManageIQ::Providers::Openstack::CloudManager::AuthKeyPair < ManageIQ::Providers::CloudManager::AuthKeyPair
+  def self.raw_create_key_pair(ext_management_system, create_options)
+    connection_options = {:service => 'Compute'}
+    ext_management_system.with_provider_connection(connection_options) do |service|
+      service.key_pairs.create(create_options)
+    end
+  rescue => err
+    _log.error "keypair=[#{name}], error: #{err}"
+    raise MiqException::MiqOpenstackApiRequestError, err.to_s, err.backtrace
+  end
+
+  def self.validate_create_key_pair(ext_management_system, options = {:name => '', :public_key => ''})
+    mandatory_options = [:name, :public_key]
+    if mandatory_options.any? { |option| options[option].blank? }
```

Comment:
I think we have to have check without data here

this check will be used for displaying the button, not before the action.

For validaton before action, we will need to propose another method for validation, we should check the code if soemthing like this is being used